src/main/java/it/pagopa/ecommerce/commons/redis/reactivetemplatewrappers/ReactiveRedisTemplateWrapper.java

```diff
@@ -0,0 +1,337 @@
+     */
+    public Mono<Boolean> save(
+                              V value,
+                              Duration ttl
+    ) {
+        return reactiveRedisTemplate.opsForValue().set(compoundKeyWithKeyspace(getKeyFromEntity(value)), value, ttl);
+    }
+
+    /**
+     * Save key to hold the string value if key is absent (SET with NX).
+     *
+     * @param value the entity to be saved
+     * @return returns false if it already exists, true if it does not exist.
+     */
+    public Mono<Boolean> saveIfAbsent(
+                                      V value
+    ) {
+        return reactiveRedisTemplate.opsForValue()
+                .setIfAbsent(compoundKeyWithKeyspace(getKeyFromEntity(value)), value, getDefaultTTL());
+    }
+
+    /**
+     * Save key to hold the string value if key is absent (SET with NX).
+     *
+     * @param value the entity to be saved
+     * @param ttl   the TTL for the entity to be saved. This parameter will override
+     *              the default TTL value
+     * @return returns false if it already exists, true if it does not exist.
+     */
+    public Mono<Boolean> saveIfAbsent(
+                                      V value,
+                                      Duration ttl
+    ) {
+        return reactiveRedisTemplate.opsForValue()
+                .setIfAbsent(compoundKeyWithKeyspace(getKeyFromEntity(value)), value, ttl);
+    }
+
+    /**
+     * Get the default configured TTL
+     *
+     * @return the default configured TTL
+     */
+    public Duration getDefaultTTL() {
+        return this.ttl;
+    }
+
+    /**
+     * Retrieve entity for the given key
+     *
+     * @param key - the key of the entity to be found
+     * @return an Optional object valued with the found entity, if any
+     */
+    public Mono<V> findById(String key) {
+        return reactiveRedisTemplate.opsForValue().get(compoundKeyWithKeyspace(key));
+    }
+
+    /**
+     * Delete the entity for the given key
+     *
+     * @param key - the entity key to be deleted
+     * @return true if the key has been removed
+     */
+    public Mono<Boolean> deleteById(String key) {
+        return reactiveRedisTemplate.delete(compoundKeyWithKeyspace(key)).map(deletedCount -> deletedCount > 0);
+    }
+
+    /**
+     * Get TTL duration for the entity with the given key Negative duration has the
+     * following meaning:
+     * <ul>
+     * <li>-1: entity that has no expiration set</li>
+     * <li>-2: input key does not exist</li>
+     * <li>-3: redis returned expiration is null</li>
+     * </ul>
+     *
+     * @param key - the entity key for which retrieve TTL
+     * @return the entity associated TTL duration.
+     */
+    public Mono<Duration> getTTL(String key) {
+        return reactiveRedisTemplate
+                .getExpire(compoundKeyWithKeyspace(key))
+                .map(ttl -> Duration.ofSeconds(ttl.isNegative() ? -3 : ttl.toSeconds()))
+                .defaultIfEmpty(Duration.ofSeconds(-3));
+    }
```

Comment:
```suggestion
   /**
     * Get TTL duration for the entity
     *
     * @param key - the entity key for which retrieve TTL
     * @return the entity associated TTL duration.
     * 
     * @see org.springframework.data.redis.core.ReactiveRedisOperations#getExpire(Object) 
     */
    public Mono<Duration> getTTL(String key) {
        return reactiveRedisTemplate
                .getExpire(compoundKeyWithKeyspace(key));
    }
```

the negative values mapping with it's explanation is valid for non reactive command. in reactive counterpart a Duration object is returned with a different semantic, so just return the framework returned value linking to it's javadoc for special value handlind (such as mono empty or mono with duration 0)